nilearn/_utils/numpy_conversions.py

```diff
@@ -168,3 +168,33 @@ def csv_to_array(csv_path, delimiters=' \t,;', **kwargs):
         array = np.genfromtxt(csv_path, delimiter=dialect.delimiter, **kwargs)
 
     return array
+
+
+def fast_same_size_concatenation(data, n_components):
+    """Utility function that return a concatenation of an array list, raising
+    an error if any array of the list has an unexpected length.
+
+    Parameters
+    ----------
+    data: list of 2D array,
+        List of arrays with the same size in first dimension.
+
+    n_components: int,
+        Expected first dimensional length
+
+    Returns
+    ----------
+    concatenated_data: 2D array,
+        Concatenated subject components
+    """
+    res = np.empty((len(data) * n_components,
```

Comment:
Why don't you take `n_components` from `data[0].shape[0]`?

Note: I think we use this kind of code pattern several time in nilearn. It shouldn't be too much work to add an `axis` parameter and use it elsewhere. If you don't plan to do it, we'll add an issue about that after merging your PR.